TestingAttributes/TestingAttribute.AttributeRetreival/RetreiveSelectiveMethods.cs

```diff
@@ -0,0 +1,117 @@
+﻿using CustomAttributes;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TestingAttribute.AttributeRetreival
+{
+    public class RetreiveSelectiveMethods
+    {
+        public List<string> SelectTestMethods(string dllPath)
+        {
+            List<string> methodList=new List<string>();
+            try
+            {
+                Assembly assembly = Assembly.LoadFrom(dllPath);
+                foreach (Type type in assembly.GetTypes())
+                {
+                    if (TestClassAttribute.Exists(type))
+                    {
+                        string className = type.FullName;
+                        
+                                foreach (MethodInfo methodInfo in type.GetMethods())
+                                {
+                                    foreach (Attribute customAttribute in methodInfo.GetCustomAttributes(false))
+                                    {
+                                        TestMethodAttribute testMethodAttribute = customAttribute as TestMethodAttribute;
+                                        if (null != testMethodAttribute)
+                                        {
+                                            methodList.Add(className + "." + methodInfo.Name);
+                                        }
+                                    }
+                                }
+                    }
+                    
+                }
+            }
+            catch (Exception e)
+            {
+                throw new System.Exception(e.Message,e);
```

Comment:
If you are just going to throw a new exception, with the message of the exception caught, why catch it at all? Catch a exception only if you want to do something there.